mlx/ops.cpp

```diff
@@ -718,6 +718,39 @@ array stack(const std::vector<array>& arrays, StreamOrDevice s /* = {} */) {
   return stack(arrays, 0, s);
 }
 
+/** array repeat with axis */
+array repeat(const array& arr, int repeats, int axis, StreamOrDevice s) {
+  axis = normalize_axis(axis, arr.ndim());
+
+  if (repeats <= 0) {
+    std::vector<int> new_shape(arr.shape());
+    new_shape[axis] = repeats > 0 ? repeats : 0;
+    return zeros(new_shape, arr.dtype(), s);
+  }
+
+  if (repeats == 1) {
+    return arr;
+  }
+
+  std::vector<int> new_shape(arr.shape());
+  new_shape[axis] *= repeats;
+  int concat_axes = arr.ndim() - axis;
+
+  std::vector<array> repeated_arrays;
+  repeated_arrays.reserve(repeats);
+
+  for (int i = 0; i < repeats; ++i) {
+    repeated_arrays.push_back(expand_dims(arr, -1, s));
+  }
+  array repeated =
+      flatten(concatenate(repeated_arrays, -1 * concat_axes, s), s);
```

Comment:
Why do you flatten the array here?

Also I think it's cleaner if you replace `-1 * concate_axes` with `axis + 1`.

Reply:
> Why do you flatten the array here?

Oh, thank you for noticing we don't really need it here,
I first implement it in python and didn't realize that I added the dim by hand, but here we have `expand_dims`.
Fixed it, what a keen eye 🥇